fix(sdk): making cron schedule more cloud agnostic

marciocadev opened the pull request:
Some characters like `?` are configurations used by AWS, but to make the schedule more cloud-agnostic, I made some changes to bring it closer to the Unix implementation

Now a cron like `* * * * *` will not throw an exception.

Closes #2849 

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

Chriscbr commented on libs/wingsdk/src/shared-aws/schedule.ts:
```diff
@@ -0,0 +1,62 @@
+  }
+
+  if (cron && dayOfMonth !== "*" && dayOfWeek !== "*") {
+    throw new Error("Cannot supply both 'day' and 'weekDay', use at most one");
```
```suggestion
    throw new Error("Cannot restrict both 'day-of-month' and 'day-of-week' in a cron expression, at least one must be '*'");
```

Chriscbr reviewed (approved):
Nice work, thanks @marciocadev !

staycoolcall911 commented:
This PR was reverted due to a failing sdk spec test: https://github.com/winglang/wing/actions/runs/8304298940/job/22729804328#step:12:1

staycoolcall911 commented:
Sorry @marciocadev - when youu get the chance can you please take a look? Happy to help and run the sdk spec tests on the fix PR

marciocadev commented:
@staycoolcall911 

The problem was that there was an error test that used to fail because the cron was only working properly for AWS. Now, I've changed it to accept the format used by UNIX, and I fixed the issue that was previously occurring. Essentially, I had to remove one of the validations because it no longer applies.
